backend/src/near.js

```diff
@@ -16,10 +21,37 @@ const queryFinalTimestamp = async () => {
 };
 
 const queryNodeStats = async () => {
-  let nodes = await nearRpc.sendJsonRpc("validators", [null]);
-  let proposals = nodes.current_proposals;
-  let currentValidators = getCurrentNodes(nodes);
-  return { currentValidators, proposals };
+  let genesisConfig = await nearRpc.sendJsonRpc(
+    "EXPERIMENTAL_protocol_config",
+    { finality: "final" }
+  );
+  let epochStatus = await nearRpc.sendJsonRpc("validators", [null]);
+  let numSeats =
+    genesisConfig.num_block_producer_seats +
+    genesisConfig.avg_hidden_validator_seats_per_shard.reduce((a, b) => a + b);
+  let currentProposals = epochStatus.current_proposals;
+  let currentValidators = getCurrentNodes(epochStatus);
+
+  if (!seatPrice) {
+    seatPrice = nearApi.validators
+      .findSeatPrice(epochStatus.current_validators, numSeats)
+      .toString();
+  }
+
+  if (!totalStake) {
+    totalStake = currentValidators
+      .reduce((acc, node) => acc.add(new BN(node.stake)), new BN(0))
+      .toString();
+  }
```

Comment:
We should update the seat price once on initialization, and then we can update it when the epoch changes (`epoch_start_height` is different from the one we queried previously)